lib/unittest.bzl

```diff
@@ -308,7 +389,31 @@ def _assert_new_set_equals(env, expected, actual, msg = None):
             full_msg = expectation_msg
         _fail(env, full_msg)
 
+def _expect_failure(env, expected_failure_msg = ""):
+    """Asserts that the target under test has failed with a given error message.
+
+    This requires that the analysis test is created with `analysistest.make()` and
+    `expect_failures = True` is specified.
+
+    Args:
+      env: The test environment returned by `unittest.begin`.
+      expected_failure_msg: The error message to expect as a result of analysis failures.
+    """
+    dep = getattr(env.ctx.attr, "target_under_test")[0]
+    if AnalysisFailureInfo in dep:
+        dep_failure = dep[AnalysisFailureInfo]
+        actual_errors = ""
+        for cause in dep[AnalysisFailureInfo].causes.to_list():
```

Comment:
And the award for most helpful PR comment goes to.....

There are fancy ways for doing this. But having said that, I'm not sure if they're more efficient/idiomatic. Nor what they are. (but I think it uses join!).

Reply:
a basic loop is fine with me if you don't feel strongly otherwise ;)